src/Cmts.ml

```diff
@@ -270,20 +270,20 @@ let partition_after_prev_or_before_next t ~prev cmts ~next =
         let same_line_as_next l =
           next.loc_start.pos_lnum = l.loc_start.pos_lnum
         in
-        let colon_before =
-          let loc_end =
-            {prev.loc_end with pos_cnum= prev.loc_end.pos_cnum - 1}
-          in
+        let char_before ~in_ =
+          let pos_cnum = prev.loc_end.pos_cnum - 1 in
+          let loc_end = {prev.loc_end with pos_cnum} in
           let char_loc = {prev with loc_start= loc_end} in
           let str = Source.string_at t.source char_loc in
-          String.equal str ";"
+          List.mem in_ str ~equal:String.equal
         in
+        let symbol_before = char_before ~in_:[";"; "|"] in
```

Comment:
Why these characters ? Eg. `&&`:
```
let () =
  xxxxxxxxxx
  (* xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx *)
  && xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
```

I don't really understand this code but why can't this be done in `Fmt_ast.ml` with more informations ? (eg. `Pexp_apply ... when is_infix_id`)

Reply:
This is done in Cmts.ml because the function deciding where the comments should be attached to (after the previous entity or before the next one) is located in this file.
Sure we can add more characters, I think maybe we could simply test if it's not an alphanumeric character (and not `)`)

Reply:
done